lib/appsignal/config.rb

```diff
@@ -90,13 +91,18 @@ def []=(key, value)
 
     def log_file_path
       path = config_hash[:log_path] || root_path
-      if path && File.writable?(path)
-        File.join(File.realpath(path), 'appsignal.log')
+      return File.join(path, 'appsignal.log') if path && File.writable?(path)
+
+      if File.writable? SYSTEM_TMP_DIR
+        $stdout.puts "appsignal: Unable to log to '#{path}'. Logging to "\
+          "'#{SYSTEM_TMP_DIR}' instead. Please check the "\
+          "permissions for the application's log directory."
+        File.join(SYSTEM_TMP_DIR, 'appsignal.log')
       else
-        '/tmp/appsignal.log'
+        $stdout.puts "appsignal: Unable to log to '#{path}' or the "\
+          "'#{SYSTEM_TMP_DIR}' fallback. Please check the permissions "\
+          "for the application's (log) directory."
       end
-    rescue Errno::ENOENT
```

Comment:
The `File.realpath` call is essential because it allows us to break out Capistrano style release directories to shared directories. Without this an agent that survives enough deploys to live through a cleanup dies.